sdk/src/quotes/public/collect-rewards-quote.ts

```diff
@@ -105,15 +121,29 @@ export function collectRewardsQuote(param: CollectRewardsQuoteParam): CollectRew
 
     // Knowing the growth of the reward checkpoint for the position, calculate and increment the amount owed for each reward.
     const amountOwedX64 = positionRewardInfo.amountOwed.shln(64);
-    rewardOwed[i] = amountOwedX64
+    const amountOwed = amountOwedX64
       .add(
         MathUtil.subUnderflowU128(
           rewardGrowthInsideX64,
           positionRewardInfo.growthInsideCheckpoint
         ).mul(liquidity)
       )
       .shrn(64);
+    
+    const transferFeeExcluded = TokenExtensionUtil.calculateTransferFeeExcludedAmount(
+      amountOwed,
+      tokenExtensionCtx.rewardTokenMintsWithProgram[i]!,
+      tokenExtensionCtx.currentEpoch
+    );
+
+    rewardOwed[i] = transferFeeExcluded.amount;
```

Comment:
What do you think about applying a similar pattern here to the rust `TransferFeeIncludedAmount` and `TransferFeeExcludedAmount` and forcing callers to handle the amounts in a more specific way rather than callers using the naming convention in `transferFee` to determine the relationship between the amount and the fee? 

So for example,
```
export type CollectRewardsQuote = [
    TransferFeeExcludedAmount | undefined,
    TransferFeeExcludedAmount | undefined,
    TransferFeeExcludedAmount | undefined,
]
```

Reply:
I've added `TransferFeeIncludedAmount` and `TransferFeeExcludedAMount` type at https://github.com/orca-so/whirlpools/pull/147
It is used as return type of `calculateTransferFeeIncludedAmount` and `calculateTransferFeeExcludedAmount`.

I kept data type of quote objects. 
- To minimize breaking change
- The fees consumed in a trade are stored in the estimatedFeeAmount field; the transferFee field is similar one, IMO.